tests/lean/structInstExtraEta.lean

```diff
@@ -0,0 +1,16 @@
+/-! This test assures that that `toA` field is not eta expanded using the `with` pattern as it was
+  before #2478 -/
+
+structure A where
+
+structure B extends A
+
+structure C extends B
+
+def a : A := sorry
+
+def b : B := { a with }
+#print b
+
+def c : C := { a with }
+#print c
```

Comment:
One more test I think would be useful is
```lean
structure D extends B

def d : D  := { c with }
#print d
```

Reply:
That’s what you get.